dice-game/service/src/main.rs

```diff
@@ -0,0 +1,153 @@
+use map_vec::Map;
+use oasis_std::{Address, Context};
+use rand::Rng;
+
+#[derive(oasis_std::Service)]
+struct DiceGame {
+    max_score: u32,
+    num_players: u32,
+    scores: Map<Address, u32>,
+}
+
+type Result<T> = std::result::Result<T, String>;
+impl DiceGame {
+    pub fn new(_ctx: &Context, num_players: u32) -> Self {
+        Self {
+            max_score: 0,
+            num_players,
+            scores: Map::new(),
+        }
+    }
+
+    /// returns whether still in play
+    pub fn is_in_play(&self, _ctx: &Context) -> bool {
+        self.scores.len() < (self.num_players as usize)
+    }
+
+    /// returns num_players
```

Comment:
```suggestion
    /// Returns the number of players.
```
Generally, don't include the names of internal members in the public docs.